domain/src/test/java/edu/nextstep/camp/domain/CalculatorTest.kt

```diff
@@ -0,0 +1,95 @@
+package edu.nextstep.camp.domain
+
+import com.example.domain.Calculator
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import java.lang.IllegalArgumentException
+
+class CalculatorTest {
+    @Test
+    fun testAdd() {
+        //given
+        val expression = "2 + 3"
+        val delimiter = " "
+        val expected = 5
+
+        //then
+        assertThat(Calculator.evaluate(expression, delimiter)).isEqualTo(expected)
+    }
+
+    @Test
+    fun testSubtract() {
+        //given
+        val expression = "2 - 3"
+        val delimiter = " "
+        val expected = -1
+
+        //then
+        assertThat(Calculator.evaluate(expression, delimiter)).isEqualTo(expected)
+    }
+
+    @Test
+    fun testMultiply() {
+        //given
+        val expression = "2 * 3"
+        val delimiter = " "
+        val expected = 6
+
+        //then
+        assertThat(Calculator.evaluate(expression, delimiter)).isEqualTo(expected)
+    }
+
+    @Test
+    fun testDivide() {
+        //given
+        val expression = "4 / 2"
+        val delimiter = " "
+        val expected = 2
+
+        //then
+        assertThat(Calculator.evaluate(expression, delimiter)).isEqualTo(expected)
+    }
+
+    @Test
+    fun evaluatesExpression() {
+        //given
+        val expression = "1 - 2 + 3 * 4 / 2"
+        val delimiter = " "
+        val expected = 4
+
+        //then
+        assertThat(Calculator.evaluate(expression, delimiter)).isEqualTo(expected)
+    }
+
+    @Test
+    fun testNonArithmeticOperatorInput() {
+        //given
+        val expression = "2 $ 3"
+        val delimiter = " "
+
+        //then
+        assertThrows<IllegalArgumentException> { Calculator.evaluate(expression, delimiter) }
```

Comment:
Exception을 테스트하는 이곳도 When Then이 합쳐진 모양새입니다.
이때에는 exception 인스턴스를 따로 받아서 처리하면 명시적으로 when then을 나눠 작성할 수 있습니다. 🙂

```suggestion
        //then
        val actualException = runCatching { Calculator.evaluate(expression, delimiter) }.exceptionOrNull()
        
        // then
        assertThat(actualException).isInstanceOf(IllegalArgumentException::class.java)
        assertThat(actualException).hasMessageThat().contains("...")
```